File: xml-and-annotation/src/main/scala/com/stulsoft/pspring/scala/annotation/based/AppConfig.scala
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.pspring.scala.annotation.based

import javax.inject.Inject
import org.slf4j.LoggerFactory
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.ApplicationContext

/** Usage of annotation-based configuration
  * @author Yuriy Stul
  */
@SpringBootApplication(scanBasePackages = Array("com.stulsoft.pspring.scala"))
class AppConfig @Inject()(ctx: ApplicationContext) extends CommandLineRunner {
  private val logger = LoggerFactory.getLogger(classOf[AppConfig])

  override def run(args: String*): Unit = {
    logger.info("==>run")
    logger.info("<==run")
  }
}

File: lombok/lombok-spring1/app/src/main/java/com/stulsoft/pspring/lombok/AppConfig.java
package com.stulsoft.pspring.lombok;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.context.annotation.Configuration;

@Slf4j
@Configuration
@RequiredArgsConstructor(onConstructor_ = @Autowired)
@EnableAutoConfiguration
public class AppConfig {
    @Value("${app.param1}")
    private Integer param1;

    public void showConfig(){
        log.info("param1 = {}", param1);
    }
}
